extensions/data-workspace/src/main.ts

```diff
@@ -16,10 +16,15 @@ import { ProjectDashboard } from './dialogs/projectDashboard';
 
 export async function activate(context: vscode.ExtensionContext): Promise<IExtension> {
 	const workspaceService = new WorkspaceService(context);
-	await workspaceService.loadTempProjects();
-	await workspaceService.checkForProjectsNotAddedToWorkspace();
-	context.subscriptions.push(vscode.workspace.onDidChangeWorkspaceFolders(async () => {
-		await workspaceService.checkForProjectsNotAddedToWorkspace();
+
+	// this is not being awaited to not block the rest of activate function from running while loading any temp projects and
+	// checking for projects not added to the workspace yet
+	workspaceService.loadTempProjects().then(() => {
+		return workspaceService.checkForProjectsNotAddedToWorkspace();
+	});
```

Comment:
where did the error handler go? :)

Reply:
I removed it since the error was already getting logged in the console, but added it back when I added the error notification